### PR TITLE
Feat: Display similar projects on project page

### LINKED ISSUE #510: Implement Sidebar for contact details and recomended projects
## Problem
Users aren't really incentivized to look at more projects when they visit a project page. They have to go back to the main page and then open another project. 

## Feature overview
We want to show similar projects on the side of the project page to show the user other content that might be interesting to them. This way it's more likely that they would find a project that they'd like to work with, exchange knowledge  with or get active in.

## Algorithm
A simple algorithm for showing similar projects would be sufficient for the start.  Information we have that could indicate that projects are similar:
- project creator (Same ProjectParent)
- location
- category (Same ProjectTags parent tag)
- rating
- (recent activity)
- (user location?)

As there are some similarities to the ClimateMatch, it could make sense to look at its [`sort_user_preferences`](https://github.com/climateconnect/climateconnect/blob/master/backend/climate_match/utility/sort_resources.py) function and see if parts of that can be reused for this. One important difference is that the ClimateMatch algorithm doesn't have to be as fast and is only accessed once for every combination of answers. The similar projects algorithm is run on every visit to any project's page so performance is more important here.

## API endpoint
There should be an endpoint that takes a project url_slug as an input and returns (5?) similar projects.

## Frontend
Similar projects should be shown in the sidebar on the project page. For the design we can use [this design](https://xd.adobe.com/view/263c33d4-4ee4-441b-7c68-f4533c94f5ee-500d/screen/bdf48a4e-501b-482a-a052-24f94e9e1baa/). However, I would remove the line around the projects as well as the different background colors and just let them float there. On mobile, they should appear under the content.
It probably makes sense for similar projects to be lazy-loaded once the rest of the page has loaded.

### PR DESCRIPTION
**Pull request by JoshKornfeld**
## Description
Closes #510 

- Changes to dependencies: Added pandas 1.4.3 and updated pytz from version 2019.3 to 2020.1.
- Integrated backend function _(implementation seen here https://github.com/climateconnect/climateconnect/pull/897)_ to get 5 similar projects.
- In the frontend, these 5 similar projects are displayed on a toggle-able sidebar on the right hand side of the screen. At the bottom of the list the user is provided a link which when clicked directs to them /browse. The sidebar is toggle-able via a button on the side of the users screen.  For smaller screen sizes, similar projects are displayed at the bottom of each tab via a existing component `ProjectPreviews`. At the top of the list, the user can minimize the list or display it. At the bottom of the list the user can go directly to /browse and view all existing projects.
- Some buttons e.g. Follow / Contact are adjusted based on whether the sidebar is displayed or screen size.
- `FollowButton.js `refactored to make it more generic

## Test plan

1. Head a page of a project
2. Sidebar should be visible and displayed by default
3. Click the button "hide similar projects" to hide it
4. Click the button "show similar projects" to display it
5. To see small screen size content, reduce screen width to <960px

## Before landing

1. PR has meaningful title
1. `yarn lint` passes (frontend)
1. `yarn format` passes (frontend)
1. `make format` passes (backend)

### CODE REVIEW

File: frontend/src/components/project/ProjectSliderWithTitle.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      paddingBottom: theme.spacing(2),
+    },
+    divider: {
+      marginTop: theme.spacing(1),
+      marginBottom: theme.spacing(1),
+    },
+  };
+});
+
+export default function ProjectSliderWithTitle({ similarProjects, title }) {
```
Comment (positiveimpact) [inline frontend/src/components/project/ProjectSliderWithTitle.js:19]: What's the advantage of an additional wrapper component over adding a `title` prop to `ProjectsSlider` directly? I'm not sure yet what's better so I'm wondering what your thought process was.

**Review by positiveimpact — CHANGES_REQUESTED** [review]
Really good job overall, but still some things to improve on! I left some comments, the biggest thing is really making sure that you don't rewrite the same code multiple times but instead make it generic (I know in the legacy code it's sometimes also not done correctly ). 
Since we have the similar projects all the way on the bottom on mobile now I think it would be better to just display them below each other. That makes it a bit more clear that there is more projects to look at and there is no other content coming anyways.

File: frontend/src/components/project/ProjectSliderWithTitle.js (hunk 1 above)
Comment (JoshKornfeld) [reply frontend/src/components/project/ProjectSliderWithTitle.js:19]: At first, I had just added a Typography + ProjectsSlider component to every tab but noticed I'm doing the same thing 3 times so I wanted to move it into a new reusable component, but I agree, adding a title prop to the slider might make more sense.

File: frontend/public/texts/project_texts.js [hunk 2]
```diff
@@ -960,5 +960,25 @@ export default function getProjectTexts({ project, user, url_slug, locale, creat
...
+      de: "Ähnliche Projekte verstecken",
+    },
+    show_recommended_projects: {
+      en: "Show similar projects",
+      de: "Ähnliche Projekte anzeigen",
+    },
+    you_may_also_like_these_projects: {
+      en: "You may also like these projects!",
+      de: "Dir könnten diese Projekte auch gefallen!",
+    },
```
[the remark this answers is not in the document]
Comment (JoshKornfeld) [reply frontend/public/texts/project_texts.js:978]: Some are in use, but yes, some others aren't. Will be deleted.

[the remark this answers is not in the document]
Comment (JoshKornfeld) [reply frontend/public/texts/project_texts.js:978]: Some are in use, some others aren't and will be deleted.

**Review by positiveimpact — CHANGES_REQUESTED** [review]
Thanks for implementing so many of the changes! There is still a few things that need to be fixed before this PR can be merged.

**Review by positiveimpact — APPROVED** [review]
LGTM now! Great work on implementing all the feedback :ok_hand: This PR is really huge, looking forward to seeing it in production :rocket:

**Pull request merged**